### PR TITLE
[vividus] Update configuration of ITERATING table transformer

### PR DESCRIPTION
**Pull request by IBrun**

### CODE REVIEW

File: vividus-tests/src/main/resources/story/integration/TableTransformers.story [hunk 1]
```diff
@@ -54,6 +54,20 @@ Scenario: Verify the number of actual iterations
...
 
 
+Scenario: Reset counter
+When I initialize the STORY variable `updatedNumber` with value `0`
+
+
+Scenario: Update the counter through ITERATING
+When I initialize the STORY variable `updatedNumber` with value `#{eval(${updatedNumber} + 1)}`
+Examples:
+{transformer=ITERATING, startInclusive=0, endInclusive=3}
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/TableTransformers.story:64]:
```suggestion
{transformer=ITERATING, startInclusive=1, endInclusive=3}
```
to verify changed behavior

File: vividus-tests/src/main/resources/story/integration/TableTransformers.story [hunk 2]
```diff
@@ -54,6 +54,20 @@ Scenario: Verify the number of actual iterations
 Then `${updatedNumber}` is = `${iterationLimit}`
 
 
+Scenario: Reset counter
+When I initialize the STORY variable `updatedNumber` with value `0`
+
+
+Scenario: Update the counter through ITERATING
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/TableTransformers.story:61]: Please mark the old scenario ad deprecated

File: vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java [hunk 3]
```diff
@@ -20,23 +20,55 @@
...
+        }
+        else
+        {
+            int startInclusive = properties.getMandatoryNonBlankProperty(START_INCLUSIVE, int.class);
+            int endInclusive = properties.getMandatoryNonBlankProperty(END_INCLUSIVE, int.class);
+            Validate.isTrue(endInclusive >= startInclusive, "'%s' value must be smaller or equal to '%s' value",
+                    START_INCLUSIVE, END_INCLUSIVE);
+            Validate.isTrue(startInclusive >= 0, "'%s' and '%s' values must be non-negative", START_INCLUSIVE,
+                    END_INCLUSIVE);
+            column = Stream.iterate(startInclusive, i -> i <= endInclusive, i -> i + 1)
```
Comment (ikalinin1) [inline vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java:68]: can we reuse generating code?

File: vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java [hunk 4]
```diff
@@ -20,23 +20,55 @@
...
+                    .map(String::valueOf)
+                    .collect(Collectors.toList());
+        }
+        else
+        {
+            int startInclusive = properties.getMandatoryNonBlankProperty(START_INCLUSIVE, int.class);
+            int endInclusive = properties.getMandatoryNonBlankProperty(END_INCLUSIVE, int.class);
+            Validate.isTrue(endInclusive >= startInclusive, "'%s' value must be smaller or equal to '%s' value",
+                    START_INCLUSIVE, END_INCLUSIVE);
+            Validate.isTrue(startInclusive >= 0, "'%s' and '%s' values must be non-negative", START_INCLUSIVE,
```
Comment (ikalinin1) [inline vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java:66]: Why do we need this constraint?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java [hunk 5]
```diff
@@ -20,21 +20,50 @@
...
+            endInclusive = limit - 1;
+        }
+        else
+        {
+            startInclusive = properties.getMandatoryNonBlankProperty(START_INCLUSIVE, int.class);
+            endInclusive = properties.getMandatoryNonBlankProperty(END_INCLUSIVE, int.class);
+            Validate.isTrue(endInclusive >= startInclusive, "'%s' value must be smaller or equal to '%s' value",
+                    START_INCLUSIVE, END_INCLUSIVE);
+        }
+        List<String> column = Stream.iterate(startInclusive, i -> i <= endInclusive, i -> i + 1)
```
Comment (uarlouski) [inline vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java:66]: why not to use IntStream.range() ?

File: docs/modules/commons/pages/table-transformers.adoc [hunk 6]
```diff
@@ -110,19 +110,27 @@ the columns.
...
 |Description
 
 |`limit`
-|threshold value, so values are generated from 0 to (`limit` - 1)
+|WARNING: The property is deprecated and will be removed in VIVIDUS 0.6.0. Use `startInclusive` and `endInclusive` instead
+
+threshold value, so values are generated from 0 to (`limit` - 1).
+
+|`startInclusive`
+|value specifying which number to start generating values from. Value should be smaller or equal to `endInclusive` value
```
Comment (valfirst) [inline docs/modules/commons/pages/table-transformers.adoc:124]:
```suggestion
|value specifying which number to start generating values from, it should be less than or equal to `endInclusive` value
```

File: docs/modules/commons/pages/table-transformers.adoc [hunk 7]
```diff
@@ -110,19 +110,27 @@ the columns.
...
+
+|`endInclusive`
+|threshold value specifying at which number to stop generating values. Value should be greater or equal to `startInclusive` value
```
Comment (valfirst) [inline docs/modules/commons/pages/table-transformers.adoc:127]:
```suggestion
|threshold value specifying at which number to stop generating values, it should be greater than or equal to `startInclusive` value
```

File: vividus-tests/src/main/resources/story/integration/TableTransformers.story [hunk 8]
```diff
@@ -40,20 +40,34 @@ Examples:
...
 
 
+Scenario: Create a counter
+Given I initialize story variable `counter` with value `0`
+
+
+Scenario: Update the counter through ITERATING with a global variable
+Given I initialize story variable `counter` with value `#{eval(${counter} + 1)}`
+Examples:
+{transformer=ITERATING, startInclusive=1, endInclusive=3}
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/TableTransformers.story:64]:
```suggestion
{transformer=ITERATING, startInclusive=1, endInclusive=$\{iterationLimit\}}
```

File: vividus-tests/src/main/resources/story/integration/VariableTests.story [hunk 9]
```diff
@@ -79,6 +79,17 @@ Then `${var#{eval(${var#{eval(<iterator> + 1)}} + 1)}}` is = `3`
 Examples:
 {transformer=ITERATING, limit=1}
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/VariableTests.story:80]:
```suggestion
{transformer=ITERATING, startInclusive=3, endInclusive=3}
```

that should be enough, no need to add a new scenario

File: vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java [hunk 10]
```diff
@@ -18,24 +18,54 @@
...
+        int startInclusive;
+        int endInclusive;
+        if (properties.getProperties().containsKey(LIMIT))
+        {
+            boolean noConflictProperties = Stream.of(START_INCLUSIVE, END_INCLUSIVE)
+                    .noneMatch(i -> properties.getProperties().containsKey(i));
+            Validate.isTrue(noConflictProperties,
+                    "Conflicting property declaration found: '%s'. Use only '%s' and '%s'", LIMIT, START_INCLUSIVE,
+                    END_INCLUSIVE);
+            LOGGER.warn("ExamplesTable property '{}' is deprecated. Use '{}' and '{}' instead", LIMIT, START_INCLUSIVE,
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java:54]: add a note that the deprecated property will be removed in VIVIDUS 0.6.0

File: vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java [hunk 11]
```diff
@@ -18,24 +18,54 @@
...
+                    END_INCLUSIVE);
+            int limit = properties.getMandatoryNonBlankProperty(LIMIT, int.class);
+            startInclusive = 0;
+            endInclusive = limit - 1;
+        }
+        else
+        {
+            startInclusive = properties.getMandatoryNonBlankProperty(START_INCLUSIVE, int.class);
+            endInclusive = properties.getMandatoryNonBlankProperty(END_INCLUSIVE, int.class);
+            Validate.isTrue(endInclusive >= startInclusive, "'%s' value must be smaller or equal to '%s' value",
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/transformer/IteratingTableTransformer.java:64]:
```suggestion
            Validate.isTrue(endInclusive >= startInclusive, "'%s' value must be less than or equal to '%s' value",
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**